frontend/src/metabase/visualizations/components/settings/ChartSettingsTableFormatting.unit.spec.js

```diff
@@ -0,0 +1,164 @@
+import { useState } from "react";
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { createMockColumn } from "metabase-types/api/mocks";
+import ChartSettingsTableFormatting from "./ChartSettingsTableFormatting";
+
+const STRING_COLUMN = createMockColumn({
+  base_type: "type/Text",
+  display_name: "String Column",
+  name: "STRING_COLUMN",
+});
+
+const STRING_COLUMN_TWO = createMockColumn({
+  base_type: "type/Text",
+  display_name: "String Column 2",
+  name: "STRING_COLUMN_2",
+});
+
+const BOOLEAN_COLUMN = createMockColumn({
+  base_type: "type/Boolean",
+  display_name: "Boolean Column",
+  name: "BOOLEAN_COLUMN",
+});
+
+const NUMBER_COLUMN = createMockColumn({
+  base_type: "type/Integer",
+  display_name: "Number Column",
+  name: "NUMBER_COLUMN",
+});
+
+const COLUMNS = [
+  STRING_COLUMN,
+  STRING_COLUMN_TWO,
+  BOOLEAN_COLUMN,
+  NUMBER_COLUMN,
+];
+
+const Wrapper = props => {
+  const [value, setValue] = useState([]);
+
+  return (
+    <ChartSettingsTableFormatting
+      cols={COLUMNS}
+      onChange={setValue}
+      value={value}
+      {...props}
+    />
+  );
+};
+
+const setup = props => {
+  render(<Wrapper {...props} />);
+};
+
+describe("ChartSettingsTableFormatting", () => {
+  it("Should allow you to add a rule", () => {
```

Comment:
The nittiest of nits

```suggestion
  it("should allow you to add a rule", () => {
```